pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -75,21 +81,26 @@ type ProviderTinkClient interface {
 	GetHardware(ctx context.Context) ([]*tink.Hardware, error)
 }
 
-func NewProvider(datacenterConfig *v1alpha1.TinkerbellDatacenterConfig, machineConfigs map[string]*v1alpha1.TinkerbellMachineConfig, clusterConfig *v1alpha1.Cluster, providerKubectlClient ProviderKubectlClient, providerTinkClient ProviderTinkClient, now types.NowFunc, skipIpCheck bool, hardwareConfigFile string) *tinkerbellProvider {
-	return NewProviderCustomNet(
+type ProviderPbnjClient interface {
```

Comment:
I don't think this interface name adequately describes the intent. Specifically, PBnJ is an implementation to achieve validating BMC credentials but as far as this interface is concerned, that point is moot. Suggest `BMCCredentialValidator` instead.

Reply:
We might use PBnJ for powering on/off machines after beta. Also, I feel we should keep the name consistent here same like ProviderTinkClient, ProviderKubectlClient, ProviderGovcClient, etc.

Reply:
I'm gonna back track here: for some reason I thought the method was non pbnj specific but it accepts a pbnj type so ignore this comment.